### DISCUSSION TITLE
Has anyone had to deal with dynamic data catalog updates?

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#5527] [Q&A]
I need to be able to update my data catalog json file programmatically with new items in a catalog group. Has anyone done this? If so - how did you do it? Need to write an API? 
Thanks,
Lisa

**Comment by hardreddata** [comment]
I have avoided doing it in Terria so far by use of https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wms-getcapabilities/ and GeoServer. The trade-off being that you need a way to get the data into GeoServer.

Good luck.
